darshan-util/pydarshan/darshan/experimental/plots/heatmap_handling.py

```diff
@@ -273,7 +273,7 @@ def get_aggregate_data(
     return agg_df
 
 
-def get_heatmap_df(agg_df: pd.DataFrame, xbins: int, nprocs: int) -> pd.DataFrame:
+def get_heatmap_df(agg_df: pd.DataFrame, xbins: int, nprocs: int, tmax: float) -> pd.DataFrame:
```

Comment:
Another design thought for consideration, again depending on what the Argonne folks really want to happen: what if `tmax` defaulted to `None`, and when it was `None`, we just return the usual properly-trimmed DXT/runtime heatmap without modification?

A bit like a sensible default that encapsulates the data--it may also have the bonus of reducing the number of testing changes apart from maybe adding a check to make sure the `tmax` is respected.

So, the plotting code could call with a custom `tmax` when it needs to make the heatmaps flush on end time, but `get_heatmap_df` still does something quite sensible even outside that scenario.

Reply:
Something like this makes sense to me. Would we want to do something similar for the minimum time? We could improve the resolution that way as well.